scheduler/cron.go

```diff
@@ -0,0 +1,198 @@
+package scheduler
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+type cron struct{}
+
+// Enable writes Fluidkeys' cron lines into crontab
+func (c *cron) Enable() (crontabWasAdded bool, err error) {
+	crontab := &systemCrontab{}
+	crontabWasAdded, err = c.enable(crontab)
+	if err != nil {
+		return crontabWasAdded, ErrModifyingCrontab{origError: err}
+	}
+	return crontabWasAdded, nil
+}
+
+// Disable parses the crontab (output of `crontab -l`) and removes Fluidkeys'
+// cron lines if present.
+// If the remaining crontab is empty, the crontab is removed with `crontab -r`
+func (c *cron) Disable() (cronLinesWereRemoved bool, err error) {
+	crontab := &systemCrontab{}
+	cronLinesWereRemoved, err = c.disable(crontab)
+	if err != nil {
+		return cronLinesWereRemoved, ErrModifyingCrontab{origError: err}
+	}
+	return cronLinesWereRemoved, nil
+}
+
+func (c *cron) Name() string {
+	return "crontab"
+}
+
+func (c *cron) enable(crontab runCrontabInterface) (crontabWasAdded bool, err error) {
+	currentCrontab, err := crontab.get()
+	if err != nil {
+		return false, fmt.Errorf("error getting crontab: %v", err)
```

Comment:
This error, for example, should probably not return `ErrModifyingCrontab`

Reply:
👍